midend/simplifyBitwise.cpp

```diff
@@ -0,0 +1,98 @@
+#include "simplifyBitwise.h"
+
+namespace P4 {
+
+bool SimplifyBitwise::Scan::preorder(const IR::AssignmentStatement *as) {
+    if (!as->right->is<IR::BOr>())
+        return false;
+    can_be_changed = true;
+    mpz_init(total_mask.get_mpz_t());
```

Comment:
I think this is not the correct way to initialize a `mpz_class` (it overwrites an already initialized mpz_t object, which would leak memory, but is innocuous due to garbage collection).  It should just be

    total_mask = 0;